endpoints/openrtb2/auction.go

```diff
@@ -1230,6 +1234,28 @@ func validateDevice(device *openrtb2.Device) error {
 	return nil
 }
 
+func validateOrFillChannel(requestPrebid *openrtb_ext.ExtRequestPrebid, requestExt *openrtb_ext.RequestExt, reqWrapper *openrtb_ext.RequestWrapper, isAmp bool) error {
+	if requestPrebid != nil {
+		if requestPrebid.Channel != nil {
+			if requestPrebid.Channel.Name == "" {
+				return errors.New("ext.prebid.channel.name can't be empty")
+			}
+		} else {
```

Comment:
This looks like it will add the channel object only if `ext.Prebid.Channel` is nil but what about when `ext.Prebid` itself is nil? We'd still want to add the channel info in the request for app and amp, no?

Reply:
We don't need the `requestPrebid` test, as the caller already verifies that it is not nil before calling this function. On making this test more aggressive, we also want to make sure we don't invalidate all non-video requests do to a missing channel name. Where is the logic to determine if a channel is needed if we are not basing it on the existence of a `Channel` object?

Reply:
Hans, for my own understanding, are you saying I don't need to have the check of if ```requestPrebid != nil```, because that's already done prior to this function call? 

And I'm confused about your second point about making the test more aggressive, are you referring to the channel support tests in ```auction_test.go```, or referring to this functions conditionals?

Reply:
Yes, it is done prior. And by making it more aggressive, I mean this validation code. Video calls may need a channel object, but I am pretty sure non-video requests don't require a channel entry. We don't want to fail valid non-video requests for the lack of a video channel, so we will want to confirm that a simple banner request will still pass as valid. With the current logic, the assumption was that if the `Channel` object exists, then a channel name is required. What will be the new "if" part that leads us to saying a channel name is required?

Reply:
About `ext.Prebid`, my point wasn't about testing its existence to avoid a nil panic. It was about adding the channel object in the request when it doesn't exist. With the current logic we're adding the channel object only if `ext.Prebid.Channel` is not present but we should be adding the channel object also if the request doesn't have `ext.Prebid` to start with. I hope that makes it a little clearer.

About your second point Hans, the channel object in not required for any type of request but if one exists, we expect the channel name to be present. We're not invalidating requests based on the presence of the channel object but rather on the presence of the name in the channel object _if_ the channel object is present. I am pasting the original proposal below. Please see the 1st and 4th point. Hopefully that provides some clarification.

> Proposal:

> Channel is an object that contains a name (required) and a version (optional)
Update the PBJS pbsBidAdapter to pass ext.prebid.channel: {name: "pbjs", version: "4.1"} unless overridden by s2sConfig.extPrebid
PBS gets the channel name out of the request if it's there
If the channel isn't present:
4a. set channel name to 'amp' if the request came in on /openrtb2/amp
4b. set channel name to 'app' if the request contains an 'app' object
4c. otherwise leave channel blank. Analytics adapters can default if they'd like.

I hope that helps!